src/models/dynamic_branch.jl

```diff
@@ -3,14 +3,18 @@ mutable struct DynamicBranch <: ACBranch
     n_states::Int64
     states::Vector{Symbol}
     internal::IS.InfrastructureSystemsInternal
-    function DynamicBranch(branch::T) where {T <: ACBranch}
+
+    function DynamicBranch(
+        branch::T;
+        internal = IS.InfrastructureSystemsInternal(),
```

Comment:
@jd-lara  In order for deserialization to work we have to be able to create the object with an existing InfrastructureSystemsInternal.  If you create an inner constructor then you must allow for that.

Reply:
Sorry, forgot about this. This needs documentation